### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by PrismTanker**
Added s4641356_OASIS_StyleGAN to recognition folder. This folder contains my solution to the problem of generating images based on the OASIS dataset using a specialised variant of the StyleGAN model.

The submitted folder contains the following:
- `model`: folder containing StyleGAN model with pretrained weights, able to be used to generate images of the OASIS dataset.
- `resources`: folder containing images displayed in `README`.md
- `GANutils.py`: contains a selection of helper functions pertaining to data pre/post processing and model training.
- `README.md`: readme, contains detailed overview of submission
- `dataset.py`: contains `OASIS_loader` class used to load and pre-process OASIS images
- `history.csv`: contains training loss history of provided pretrained model, used to generate loss plots
- `modules.py`: contains custom subclassed keras layers and the styleGAN class representing a styleGAN model
- `predict.py`: contains the `generate_images` function to generate images of the OASIS dataset using trained a styleGAN model
- `train.py`: contains `train` function and associated `fit` helper function used to train a styleGAN model on sample images

### CODE REVIEW

**Comment by SiyuLiu0329** [comment]
**This is an initial inspection, no action is required at this point**
Readme: Good
Results: OK
Code: OK
Commit messages: Good

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (minor performance issue) -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove CSV and pb files -2
Request Description good

**Comment by PrismTanker** [comment]
Acted on feedback:
Removed CSV and pb files
